### ISSUE TITLE
Transaction and Operation Cursor resend records after a period of time

### ISSUE THREAD

**Issue by jackmull** [elucidsoft/dotnet-stellar-sdk#27]
I have been modifying Program.cs in TestConsole to try out different SDK functions.

While testing Cursors I entered the following command:
           server.Transactions
                .Cursor("39074633206480896")
                .Limit(20)
                .ForAccount(KeyPair.FromAccountId(testAccount))
                .Order(OrderDirection.ASC)
                .Stream((sender, response) => { ShowTransactionRecord(response, _server); })
                .Connect();

That's a valid cursor and account.
Transactions are displayed on the console. Then after a minute the following exception is thrown.
![image](https://user-images.githubusercontent.com/7691125/40518113-e32e364c-5f86-11e8-82cf-10c35511229c.png)

And the records are again sent to the callback method and displayed on the console:
![image](https://user-images.githubusercontent.com/7691125/40518371-4b6c2132-5f88-11e8-849e-d9120dbc78ac.png)

**Comment by elucidsoft** [comment]
Yeah this is a horizon issue, it's behind a load balancer which disconnects the stream after a period of time. You shouldn't see this behavior if you stand up your own node and use that.

**Comment by jackmull** [comment]
Thanks for the reply Eric. I am going to stand up my own node. But even
then, on the rare occasion that the stream has to be re-connected, the code
will send a WebRequest with the original cursor which will replay the
records. Is there a way to add a callback so the client can bump the cursor
to the last record he processed?

Thanks, Jack

**Comment by elucidsoft** [comment]
Not sure, off the top that sounds like it would be pretty difficult. Currently there is not State tracking done, so there is no way to know what the last record was prior to the disconnect. You could build out an additional layer for your scenario and have it keep track and on disconnect you could restart the cursor. We could put more events into the even streaming system to allow you to have more control and do that.

**Comment by jackmull** [comment]
Yeah, there may not be an elegant solution to this.
If the Url could change with each event that might work.
In particular, the ?cursor=x part could change to ?cursor=x'.
Then if a re-connect is needed the WebRequest would have the correct Url.
A bit kludgey though.
